jalib/jassert.cpp

```diff
@@ -46,6 +43,9 @@ int jassert_quiet = 0;
 static int theLogFileFd = -1;
 static int errConsoleFd = -1;
 
+static jalib::string& tmpDir() {static jalib::string s;return s;};
+static jalib::string& uniquePidStr() {static jalib::string s;return s;};
```

Comment:
I'm not proposing to mix C and C++ styles.  I just want to limit the code to a simple enough C++ style that new developers can understand the code without too much effort.  You wrote "This whole library uses heavy C++ so I wouldn't worry about the intricacies."  I found the rest of the code in the file very readable.
In this case, I'm speaking from my experience at working with many students, and observing what they do understand and what they do not understand.  If we ask students to add code to DMTCP using a style that they do not understand, we are asking for trouble.  The convention used here is not taught in standard C++ courses.  _Isn't there any common convention in C++ that would be understood by students who have had only C, Java, and perhaps one semester of C++?_

If this conversation goes further, let's talk on the phone rather than messaging.  But the issue that students face is that the statement `tmpDir() = ...` does not have any mention within the current function of what the destination type is.  Students know that C++ will do appropriate type conversions, but that's a hard concept to apply when the destination type is not specified.